### ISSUE TITLE
Detect typos in `in` checks for TypedDicts

### ISSUE THREAD

**Issue by not-my-profile** [microsoft/pyright#4376]
```python
# pyright: strict

from typing import TypedDict, NotRequired, Literal

def foo(num: int, map: dict[str, str]):
    # pyright already detects the following:
    if num in map: # error: Expression will always evaluate to False since the types "int" and "str" have no overlap
        pass

class Document(TypedDict):
    licensed: NotRequired[Literal[True]]

def check(doc: Document):
    # however the following sneaky typo is not caught by pyright:
    if 'licenced' in doc:
        pass
```

Such `in` checks are quite common with `TypedDict`s that have optional fields (either via `total=False` or via `NotRequired`), so it would be great if Pyright could also detect such typos (probably only in strict mode due to the performance overhead).

**Comment by erictraut** [comment]
Pyright already contains checks for unguarded access to TypedDict fields. For example, if you attempt to access `doc['licensed']` in the `if` block in your example above, it will report that error. I think the check you're suggesting here would be largely redundant.

```python
def check(doc: Document):
    if 'licenced' in doc:
        x = doc['licensed'] # Error: 'licensed' is not a required key
```

**Comment by not-my-profile** [comment]
Right, this is especially problematic for optional `TypedDict` entries that have a `Literal[True]` value because with such fields you are unlikely to access the value after checking for its presence since you already now the value.

Good point on the TypedDict in my example needing to be final for the example to make sense ... I'll amend my example to add `@typing.final` to the class.
